src/java.base/share/classes/sun/security/ssl/CertificateMessage.java

```diff
@@ -1077,23 +1079,18 @@ private static SSLPossession choosePossession(
                     checkedKeyTypes.add(ss.keyAlgorithm);
                     continue;
                 }
-
-                SSLPossession pos = ka.createPossession(hc);
-                if (pos == null) {
-                    if (SSLLogger.isOn && SSLLogger.isOn("ssl,handshake")) {
-                        SSLLogger.warning(
-                            "Unavailable authentication scheme: " + ss.name);
-                    }
-                    continue;
-                }
-
-                return pos;
+                allAuths.add(ss.keyAlgorithm);
             }
 
-            if (SSLLogger.isOn && SSLLogger.isOn("ssl,handshake")) {
-                SSLLogger.warning("No available authentication scheme");
+            SSLPossession pos = X509Authentication
+                    .createPossession(hc, allAuths.toArray(String[]::new));
+            if (pos == null) {
+                if (SSLLogger.isOn && SSLLogger.isOn("ssl,handshake")) {
+                    SSLLogger.warning(
+                            "Unavailable authentication scheme: " + allAuths);
```

Comment:
The message may be missleading as it only shows a part of the request authentication scheme.  It is different from the previous code, which check the scheme one by one.  I may just log the message as:

`SSLLogger.warning("No available authentication scheme");`

Reply:
Sure.